src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -84,7 +81,8 @@ export const Deposit: React.FC<DepositProps> = ({
   fromAddress,
   onNext,
 }) => {
-  const [isApprovalRequired, setIsApprovalRequired] = useState(false)
+  // Redeclared here as state field because of chicken-and-egg issues
+  const [isApprovalRequired, setIsApprovalRequired] = useState<boolean | undefined>(false)
```

Comment:
q: what is the issue? I feel like this is not needed at all

Reply:
https://github.com/shapeshift/web/pull/7670#discussion_r1743959147